torch/csrc/jit/codegen/cuda/compute_at.cpp

```diff
@@ -69,8 +70,11 @@ unsigned int getReplayablePosPasC(
   // not be inlined to vectorized dimensions in consumer.
   auto c_dom = consumer->domain()->domain();
   auto vector_dim_it =
-      std::find_if(c_dom.begin(), c_dom.end(), [](IterDomain* id) {
-        return isParallelTypeVectorize(id->getParallelType());
+      std::find_if(c_dom.begin(), c_dom.end(), [&mode](IterDomain* id) {
+        return isParallelTypeVectorize(id->getParallelType()) ||
+            ((mode == ComputeAtMode::BestEffort ||
+              mode == ComputeAtMode::MostInlined) &&
+             id->getParallelType() == ParallelType::Unroll);
```

Comment:
I mostly stole unroll here so I could modify where compute at goes throughout the DAG. Likely we should have some kind of markers that will indicate on a per tensor view basis to not compute at inside that position.

Reply:
Can you please explain why the replay position is limited by unrolled IDs? Also, is it only for unroll? What about unswitch?

Reply:
This was a convenience mechanism and a bit of a hack for now. Right now we don't have detailed control of inlining during compute at. It's really all or nothing, and even if we try to separate this out into multiple compute at calls, because of the common consumer logic often it will go over sections you don't intend. I really just did this so I could prevent inlining at some finer grained targeted positions. Longer term we will likely need to have some markings on a per tensor basis. This challenging to do now because I'm uncertain what we will need in the mid term and I don't want to over engineer it for something we need short term.

Reply:
A little bit more here. Primarily what was happening was the unrolling dimension was getting inlined when storing to gmem. This happens frequently because in multi output cases compute at frequently propagates to all outputs. So if we do a most inlined to something stopped short of the output, it will inline the storing of the outputs. By explicitly marking them as unrolled and nothing else, it will prevent this from happening. I do it https://github.com/csarofeen/pytorch/pull/1035/files#diff-b3b08eee003a4d91eb8b8ce8e789fdec413a0442c5a8434a03532f29f84a361bR1172-R1187 primarily for the following https://github.com/csarofeen/pytorch/pull/1035/files#diff-b3b08eee003a4d91eb8b8ce8e789fdec413a0442c5a8434a03532f29f84a361bR1258-R1263